agent/http.go

```diff
@@ -210,7 +210,14 @@ func (s *HTTPHandlers) handler(enableDebug bool) http.Handler {
 	// handlePProf takes the given pattern and pprof handler
 	// and wraps it to add authorization and metrics
 	handlePProf := func(pattern string, handler http.HandlerFunc) {
+
 		wrapper := func(resp http.ResponseWriter, req *http.Request) {
+
+			// If enableDebug or ACL enabled, register wrapped pprof handlers
+			if !s.agent.config.EnableDebug && s.checkACLDisabled() {
+				return
```

Comment:
Maybe we can write a 404 in the `responseWriter` to avoid introducing a breaking change.